### PR TITLE
Custom hooks

### PR DESCRIPTION
**Pull request by brunomichetti**
- Create custom hooks
- Remove useEffect and useState from components

### CODE REVIEW

File: src/components/users/EditProfileForm.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
 } from 'constants/user.constants';
 import CustomLoader from 'components/common/CustomLoader';
 import { editProfileConstraints } from 'helpers/users-constraints';
+import useEditProdile from 'hooks/useEditProfile';
```
Comment (tamarasuarezrod) [inline src/components/users/EditProfileForm.js]:
```suggestion
import useEditProfile from 'hooks/useEditProfile';
```

File: src/components/users/EditProfileForm.js [hunk 2]
```diff
@@ -27,9 +28,7 @@ const EditProfileForm = ({ user, setEditProfile, newImg }) => {
 
   const [errors, setErrors] = useState({});
 
-  const { loading, requestError, errorMsg, updated } = useSelector(
-    (state) => state.user
-  );
+  const { loading, requestError, errorMsg } = useEditProdile(setEditProfile);
```
Comment (tamarasuarezrod) [inline src/components/users/EditProfileForm.js]:
```suggestion
  const { loading, requestError, errorMsg } = useEditProfile(setEditProfile);
```

**Review by tamarasuarezrod — APPROVED** [review]
🤩

**Review by manema — APPROVED** [review]
Great!

**Pull request merged**